sdk/go/common/resource/plugin/analyzer_plugin.go

```diff
@@ -611,6 +611,23 @@ func (a *analyzer) Close() error {
 	return a.plug.Close()
 }
 
+// Cancel signals the analyzer to gracefully shut down and abort any ongoing analysis operations.
+func (a *analyzer) Cancel(ctx context.Context) error {
+	label := a.label() + ".Cancel()"
+	logging.V(7).Infof("%s executing", label)
+
+	_, err := a.client.Cancel(a.ctx.Request(), &emptypb.Empty{})
```

Comment:
Should we not just pass `ctx` here? Not sure what the pattern is elsewhere but feels like we should use the argument in some fashion.